### ISSUE TITLE
Amortissement et déphasage

### ISSUE THREAD

**Issue by Ewen-Marzin** [flipoyo/MOLONARI1D#205]
Analyse d'amortissement et de déphasage avec les transformations de Fourier

**Comment by vargoclapin** [comment]
Fonction polynomiale

```python
from sklearn.metrics import r2_score

def rss_score(y_true, y_pred):
    return np.sum((y_true - y_pred)**2)


def regression_poly(reg, deg, show_plot=False):
    #fft puis ln amplitude et on a les 4 points dans le tableau reg
    coeffs = np.polyfit(depth_sensors, reg, deg)
    p = np.poly1d(coeffs)

    # Prédictions
    y_pred = p(depth_sensors)
    r2 = r2_score(reg, y_pred)
    if not show_plot:
        return r2, rss_score(reg, y_pred)
    else:
        plt.scatter(depth_sensors, reg, label='Données FFT mesurées')
        plt.plot(depth_sensors, y_pred, 'r--', label=f'Ajustement poly {deg}: R²={r2:.3f}')
        plt.xlabel('Profondeur z (m)')
        plt.ylabel('Log amplitude ln(A(z))')
        plt.title(f"Régression polynomiale de degré {deg}, Le r carré vaut {r2:.3f}")
        plt.show()
        return r2, rss_score(reg, y_pred)

    

def critere(reg,t_sigma):
    vraisemblance = []
    for deg in range(1, 3):
        _, rss = regression_poly(reg, deg)
        L = -0.5 * len(depth_sensors) * np.log(2*np.pi*t_sigma**2)- 0.5*rss/(t_sigma**2)
        vr = 2*deg - 2*L+(2*deg)*(deg+1)/(len(depth_sensors)-deg)
        vraisemblance.append(vr)
    min = np.inf
    for i in range(len(vraisemblance)):
        if vraisemblance[i] < min:
            min = vraisemblance[i]
            deg = i + 1
    if deg == 1:
        return("Modèle linéaire et donc modèle 1D validé préféré selon le critère d'information d'Akaike corrigé (AICc).", vraisemblance)
    else:
        return("Modèle non linéaire et donc modèle 2D préférable selon le critère d'information d'Akaike corrigé (AICc).", vraisemblance)


        

r2, rss = regression_poly(log_amplitude_ratio, 1, show_plot=True)
r2, rss = regression_poly(log_amplitude_ratio, 2, show_plot=True)

print(critere(log_amplitude_ratio, 0.3))

```

**Comment by PyNoe** [comment]
## Liste des choses à faire pour la semaine prochaine.

- Regarder avec un seul sinus et en faisant attention à la charge et à prendre des capteurs un peu plus en profondeur.
- Regarder avec deux pics pour remplacer le déphasage.
- Verification du cas conductif pur (i.e $v_t = 0$) donc pas de déphasage et voir si on retrouve la diffusivité $\kappa_e$.
- Sur des données 2D de Agnès, regarder le critère 1D/2D via régression polynomiale. (regarder le solveur d'Adam branche 226 jupyter notebook.) 
- SI ça marche on fait des beaux plots pour la présentation.
- (Nettoyer le code pour la génération multipériodique).

Bises.

**Comment by PyNoe** [comment]
Le calcul du gradient hydraulique dans le code `pyheatme` est calculé via : 

```python
gradH = - dH_offset/Zbottom
```

Donc en gros `riverbed` ne sert absolument à rien
